airflow/providers/google/cloud/transfers/calendar_to_gcs.py

```diff
@@ -0,0 +1,178 @@
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import json
+from datetime import datetime
+from tempfile import NamedTemporaryFile
+from typing import Any, List, Optional, Sequence, Union
+
+from airflow.models import BaseOperator
+from airflow.providers.google.cloud.hooks.gcs import GCSHook
+from airflow.providers.google.suite.hooks.calendar import GoogleCalendarHook
+
+
+class GoogleCalendarToGCSOperator(BaseOperator):
+    """
+    Writes Google Calendar data into Google Cloud Storage.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:TODO `howto/operator:GoogleCalendarToGCSOperator`
+
+    :param calendar_id: The Google Calendar ID to interact with.
+    :type calendar_id: str
+    :param destination_bucket: The destination Google cloud storage bucket where the
+        report should be written to. (templated)
+    :type destination_bucket: str
+    :param destination_path: The Google cloud storage URI array for the object created by the operator.
+        For example: ``path/to/my/files``.
+    :type destination_path: str
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :type impersonation_chain: Union[str, Sequence[str]]
+    """
+
+    template_fields = [
+        "calendar_id",
+        "destination_bucket",
+        "destination_path",
+        "impersonation_chain",
+    ]
+
+    def __init__(
+        self,
+        *,
+        destination_bucket: str,
+        calendar_id: str = "primary",
+        i_cal_uid: Optional[str] = None,
+        max_attendees: Optional[int] = None,
+        max_results: Optional[int] = None,
+        order_by: Optional[str] = None,
+        private_extended_property: Optional[str] = None,
+        q: Optional[str] = None,
+        shared_extended_property: Optional[str] = None,
+        show_deleted: Optional[bool] = False,
+        show_hidden_invitation: Optional[bool] = False,
```

Comment:
```suggestion
        show_deleted: bool = False,
        show_hidden_invitation: bool = False,
```
The `Optional` typing implies that this value _could_ be `None`. It doesn't seem like this is the case.

Reply:
The parameter is Optional and the default value is False.

I think either suggestion of making in `None` or not `Optional` will work. I will test it out.